### PR TITLE
feat(helm chart): add value to modify ps retries

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Adds a helm value to modify the io-engine `ps-retries`.

### CODE REVIEW

File: chart/templates/mayastor/io/io-engine-daemonset.yaml [hunk 1]
```diff
@@ -110,7 +110,8 @@ spec:
         - "--env-context=--{{ .Values.io_engine.envcontext }}"{{ end }}{{ if .Values.io_engine.reactorFreezeDetection.enabled }}
         - "--reactor-freeze-detection"{{ end }}
         - "--tgt-crdt={{ .Values.io_engine.target.nvmf.hostCmdRetryDelay.crdt1 }}"{{ if .Values.eventing.enabled }}
-        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}
+        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}{{ if .Values.io_engine.ps_retries}}
+        - "--ps-retries=300"{{ end }}
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/io/io-engine-daemonset.yaml]:
```suggestion
        - "--ps-retries={{ .Values.io_engine.pstorRetries }}"{{ end }}
```

File: chart/values.yaml [hunk 2]
```diff
@@ -466,6 +466,8 @@ io_engine:
   priorityClassName: ""
   # -- Runtime class to use. Defaults to cluster standard
   runtimeClassName: ""
+  # -- Number of retries for pstor persistence before target self shutdown
+  ps_retries: "300"
```
Comment (tiagolobocastro) [inline chart/values.yaml]:
```suggestion
  pstorRetries: "300"
```

File: chart/values.yaml [hunk 3]
```diff
@@ -466,6 +466,8 @@ io_engine:
   priorityClassName: ""
   # -- Runtime class to use. Defaults to cluster standard
   runtimeClassName: ""
+  # -- Number of retries for pstor persistence before target self shutdown
```
Comment (tiagolobocastro) [inline chart/values.yaml]:
```suggestion
  # -- Number of retries for pstor persistence before the volume target self shutdowns
```

File: chart/templates/mayastor/io/io-engine-daemonset.yaml (hunk 1 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: Oops. Thanks for pointing this out.

**Review by dsharma-dc — APPROVED** [review]

File: chart/templates/mayastor/io/io-engine-daemonset.yaml [hunk 4]
```diff
@@ -110,7 +110,8 @@ spec:
         - "--env-context=--{{ .Values.io_engine.envcontext }}"{{ end }}{{ if .Values.io_engine.reactorFreezeDetection.enabled }}
         - "--reactor-freeze-detection"{{ end }}
         - "--tgt-crdt={{ .Values.io_engine.target.nvmf.hostCmdRetryDelay.crdt1 }}"{{ if .Values.eventing.enabled }}
-        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}
+        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}{{ if .Values.io_engine.pstorRetries}}
+        - "--ps-retries={{ .Values.io_engine.pstorRetries }}"{{ end }}
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/io/io-engine-daemonset.yaml]: Should we use default rather than if? For upgrades with old values?

**Review by tiagolobocastro — APPROVED** [review]

File: chart/templates/mayastor/io/io-engine-daemonset.yaml [hunk 5]
```diff
@@ -110,7 +110,8 @@ spec:
         - "--env-context=--{{ .Values.io_engine.envcontext }}"{{ end }}{{ if .Values.io_engine.reactorFreezeDetection.enabled }}
         - "--reactor-freeze-detection"{{ end }}
         - "--tgt-crdt={{ .Values.io_engine.target.nvmf.hostCmdRetryDelay.crdt1 }}"{{ if .Values.eventing.enabled }}
-        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}
+        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}{{ if .Values.io_engine.pstorRetries}}
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/io/io-engine-daemonset.yaml]:
```suggestion
        - "--events-url=nats://{{ .Release.Name }}-nats:4222"{{ end }}{{ if .Values.io_engine.pstorRetries }}
```

File: chart/templates/mayastor/io/io-engine-daemonset.yaml (hunk 4 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: Yeah we can? Should I do that now?

Comment (niladrih) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: This is new value, this should get pulled in. Upgrades via upgrade job should be fine.

File: chart/values.yaml [hunk 6]
```diff
@@ -466,6 +466,8 @@ io_engine:
   priorityClassName: ""
   # -- Runtime class to use. Defaults to cluster standard
   runtimeClassName: ""
+  # -- Number of retries for pstor persistence before the volume target self shutdowns
+  pstorRetries: "300"
```
Comment (niladrih) [inline chart/values.yaml]: This looks like an int, strange to use it as a string. Unless it makes some logical sense, e.g. the rust data type is a String.

Comment (Abhinandan-Purkait) [reply chart/values.yaml]: The parser parses it into a u16 anyway

File: chart/templates/mayastor/io/io-engine-daemonset.yaml (hunk 4 above)
Comment (Abhinandan-Purkait) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: Ohh then we don't need it? I have added it anyway. Should I remove it?

Comment (niladrih) [reply chart/templates/mayastor/io/io-engine-daemonset.yaml]: It's fine either way. Don't need to change IMO.

File: chart/values.yaml (hunk 6 above)
Comment (niladrih) [reply chart/values.yaml]: Yeah, I'd say we should remove the `""` here in that case.

Comment (Abhinandan-Purkait) [reply chart/values.yaml]: Okay.

**Review by niladrih — APPROVED** [review]

File: chart/values.yaml (hunk 6 above)
Comment (Abhinandan-Purkait) [reply chart/values.yaml]: Done.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Comment by niladrih** [comment]
> bors merge

Needs conversation resolution. This will likely fail otherwise.

**Comment by Abhinandan-Purkait** [comment]
> > bors merge
> 
> Needs conversation resolution. This will likely fail otherwise.

I think I did resolve before the comment. Did I not?

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**